Add csutter's lightweight templates

csutter opened the pull request:
I've started work on a set of devcontainer templates that are similar to, but differently opinionated from the original templates in [devcontainers/templates](https://github.com/devcontainers/templates). In particular, they are intended to be very basic/lightweight/"batteries excluded", and use upstream images for their respective ecosystem instead of the custom ones used by the default templates.

I've been a little bit on the fence about adding these to the collection of available templates because VS Code currently puts them all into one big list without categorisation, and there may be a risk of less experienced users who would be better off with the original templates getting confused.

Would love to hear some thoughts and take it from there!

csutter committed:
Add csutter's lightweight templates

- Adds `csutter/lightweight-devcontainer-templates` to the collection
  of available devcontainer templates

samruddhikhandale commented:
Hi @csutter,

Thanks for creating your own set of Templates. 

> I've been a little bit on the fence about adding these to the collection of available templates because VS Code currently puts them all into one big list without categorisation, and there may be a risk of less experienced users who would be better off with the original templates getting confused.

In my opinion, I think that should be alright. As we are trying to provide a marketplace experience for the users, the variety could also help them pick a right one according to their usage.

![image](https://user-images.githubusercontent.com/24955913/211369957-d13250c1-34a3-4908-9dcd-4f68534dcb15.png)

The VS Code lists the Templates with a short description &  ℹ️  button which would navigate to the README. The user could roughly go through the README/source code and decide.

> In particular, they are intended to be very basic/lightweight/"batteries excluded", and use upstream images for their respective ecosystem instead of the custom ones used by the default templates.

The custom MCR images used by the Templates are based on the ecosystem ones (eg. [Ruby](https://github.com/devcontainers/images/blob/main/src/ruby/.devcontainer/Dockerfile#L3)). The custom image adds  common tools/packages, themes, helps configure user, add VS Code Extensions etc. It's an attempt to provide a complete development experience when the user choses a particular Template. 

However, in some cases the user might not want this extra stuff and your templates could come in handy. So I think it's nice to get them added to the collection, would definitely benefit the community.

Do you have any other thoughts?

samruddhikhandale reviewed (changes requested):
PR looks good, can you help resolve the merge conflicts? thanks

csutter committed:
Merge branch 'gh-pages' into patch-1

csutter commented:
Thanks for your thoughts @samruddhikhandale - really appreciate your input 😄

That all sounds like what I thought - agree that the MCR images are more useful to the average user with their customisations and integration, but that there's a space for less customised bespoke templates too.

I've renamed my repository and changed the description and README to make it a bit more clear what the motivation and goals for my set of templates is, and fixed the merge conflict so hoping this should be ready to go!

samruddhikhandale reviewed (approved):
Awesome, thank you! 🚀
